server/controllers/finance/costCenter.js

```diff
@@ -102,8 +106,45 @@ function lookupCostCenter(id, codes) {
     });
 }
 
+function getCostValue (req, res, next){
```

Comment:
Could you write some comments describing what the procedure does?  It is quite complex.